### DISCUSSION TITLE
How does vcpkg find Visual Studio?

### DISCUSSION THREAD

**Discussion by thxkiwi** [microsoft/vcpkg#20961] [General]
I have a vcpkg_root that is integrated with Visual Studio 2019 installed on the system drive (C:).

I'd like to have another vcpkg_root that is integrated with a different installation of the build tools located on D:

How can I make vcpkg find and use a specific location of the toolchain?

**Comment by autoantwort** [comment]
You are looking for [VCPKG_VISUAL_STUDIO_PATH](https://github.com/microsoft/vcpkg/blob/master/docs/users/config-environment.md#vcpkg_visual_studio_path)?

**Reply by thxkiwi** [reply]
I should've found this myself.

Thank you.

**Reply by thxkiwi** [reply]
Seems like I need a little more help here...

What exactly is vcpkg looking for? Are there vcpkg logs somewhere? I know the buildtrees have logs for each package/port but I'm not sure where to look for logs of vcpkg itself.

```
C:\Users\miken\Documents\src\srcroot>dir "%VCPKG_VISUAL_STUDIO_PATH%"
 Volume in drive D is EW_MULFREO_EN-US_DV9
 Volume Serial Number is CF85-3C85

 Directory of D:\Program Files\Microsoft Visual Studio\2019\BuildTools

12/02/2020  15:47    <DIR>          .
12/02/2020  15:47    <DIR>          ..
12/02/2020  15:44                42 CHPEVersion.txt
12/02/2020  15:47    <DIR>          Common7
12/02/2020  15:47    <DIR>          DIA SDK
12/02/2020  15:47    <DIR>          Licenses
12/02/2020  15:47    <DIR>          MSBuild
12/02/2020  15:47    <DIR>          SDK
12/02/2020  15:47    <DIR>          Team Tools
12/02/2020  15:47    <DIR>          VC
12/02/2020  15:43                78 Version.txt
12/02/2020  15:43               526 asan_blacklist.txt
12/02/2020  15:43           452,488 clang_rt.asan_dynamic-i386.dll
               4 File(s)        453,134 bytes
               9 Dir(s)               0 bytes free

C:\Users\miken\Documents\src\srcroot>"C:\Users\miken\Documents\bin\vcpkg-ewdk-19041\scripts\buildsystems\msbuild\..\..\..\vcpkg.exe" install  --x-wait-for-lock --triplet "x64-windows-static" --vcpkg-root "C:\Users\miken\Documents\bin\vcpkg-ewdk-19041\scripts\buildsystems\msbuild\..\..\..\\" "--x-manifest-root=C:\Users\miken\Documents\src\srcroot\project\subproject1\\" "--x-install-root=C:\Users\miken\Documents\src\apo\project\subproject1\vcpkg_installed\\"
Could not find Visual Studio instance at D:\Program Files\Microsoft Visual Studio\2019\BuildTools.

```

**Reply by thxkiwi** [reply]
*bump* 

I'm happy to dig through a vcpkg install I'm just not sure where to begin to determine why setting `VCPKG_VISUAL_STUDIO_PATH` is not able to find a valid install. This install is definitely usable for building any C/C++ MSBUILD project.

Can anyone help me understand why vcpkg insists this is not a usable instance?

**Reply by thxkiwi** [reply]
@autoantwort any ideas why vcpkg wouldn't find a usable instance with the above? [thanks for understanding my tagging you here. Haven't got a lot of traction on this one. Grateful for any help you can offer.]

**Reply by autoantwort** [reply]
No, sorry
